pdata/internal/json/scope.go

```diff
@@ -0,0 +1,42 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json // import "go.opentelemetry.io/collector/pdata/internal/json"
+
+import (
+	jsoniter "github.com/json-iterator/go"
+
+	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
+)
+
+func ReadScope(iter *jsoniter.Iterator, scope *otlpcommon.InstrumentationScope) {
+	iter.ReadObjectCB(func(iter *jsoniter.Iterator, f string) bool {
+		switch f {
+		case "name":
```

Comment:
Any chance there are constants from the spec/proto we can use to represent these cases?

Reply:
They are tags, but that needs reflection package and is expensive. Not anything else that I know, but probably we can do a proto plugin to extract these values.

Reply:
Looking through pdata seems like we use this strategy a lot, so changing,  if possible, would probably be better for a later PR.

Reply:
Definitely will not do this in the current PR, also needs a lot of changes :)